src/components/dialog/Dialog.tsx

```diff
@@ -0,0 +1,117 @@
+"use client"
+import { useSearchParams, useRouter } from 'next/navigation'
+import { useRef, useEffect, useState } from 'react';
+
+const SEARCH_URL = "http://openlibrary.org";
+
+type Props = {
+    onClose: () => void,
+    onAddToRead: () => void,
+    onAddToWantToRead: () => void,
+    loggedIn: boolean,
+}
+
+export default function Dialog({ onClose, onAddToRead, onAddToWantToRead, loggedIn }: Props) {
+  const router = useRouter()
+
+  const searchParams = useSearchParams()
+    const dialogRef = useRef<null | HTMLDialogElement>(null)
+    const showDialog = searchParams.get('showDialog')
+    const bookId = searchParams.get('search');
+
+    const [bookData, setBookData] = useState<any>({});
+    const [authorData, setAuthorData] = useState<any>({});
+    const [loading, setLoading] = useState(false);
+
+    useEffect(() => {
+        if (showDialog === 'y') {
+            dialogRef.current?.showModal()
+        } else {
+            dialogRef.current?.close()
+        }
+    }, [showDialog]);
+
+    useEffect(() => {
+        setLoading(true);
+        if (bookId !== null ) {
+            fetch(`${SEARCH_URL}${bookId}.json`)
+            .then((res) => {
+                return res.json();
+            })
+            .then((data) => {
+                setBookData(data);
+            });
+        }
+      }, [bookId]);
+
+      useEffect(() => {
+        if (bookData.authors) {
+            fetch(`${SEARCH_URL}${bookData.authors[0].author.key}.json`)
+            .then((res) => {
+                return res.json();
+            })
+            .then((data) => {
+                setAuthorData(data);
+                setLoading(false);
+            });
+        }
```

Comment:
proposal: take fetch logic out of the component and move it to a `client` file that has all of the fetching stuff and error handling there (see:  `src/app/clients/book-client.ts` for example)

Reply:
Optional: if you want to use async/await in a useEffect there is a pattern you can follow: https://ultimatecourses.com/blog/using-async-await-inside-react-use-effect-hook

TLDR: define the async function in the useEffect and then call it right after below